subprojects/enterprise/src/main/java/org/gradle/internal/tasks/TaskProperties.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.tasks;
+
+import java.util.Set;
+
+public interface TaskProperties {
+
+    Set<InputFileProperty> getInputFiles();
+
+    Set<OutputFileProperty> getOutputFiles();
```

Comment:
I wonder if this should be a `Set` or not. True, we want to avoid duplicates, but constructing a `Set` might be an expensive operation we could avoid here?

Reply:
I don't think these would usually be particularly large sets so I don't think performance is an issue here. What would you propose?